src/mraa.c

```diff
@@ -97,6 +97,57 @@ mraa_set_log_level(int level)
     return MRAA_ERROR_INVALID_PARAMETER;
 }
 
+mraa_boolean_t mraa_is_kernel_chardev_interface_compatible()
```

Comment:
Hi. From what i understand, it's best to leave the check at runtime, in order to avoid problems like building a project separately on a host that supports the chardev support and running on a target that doesn't support it. I must agree that this function needs further work to cover all possible cases(like the one exposed by you).

Reply:
I'll probably leave this to @arfoll to decide on runtime/buildtime check. But like you said, we need to investigate on all possible cases.